src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -565,7 +565,11 @@ public async Task PingPong()
         {
             PingFrame ping = new PingFrame(new byte[8] { 1, 2, 3, 4, 50, 60, 70, 80 }, FrameFlags.None, 0);
             await WriteFrameAsync(ping).ConfigureAwait(false);
-            await ReadFrameAsync(Timeout).ConfigureAwait(false);
+            Frame pingAck = await ReadFrameAsync(Timeout).ConfigureAwait(false);
+            if (pingAck.Type != FrameType.Ping || !pingAck.AckFlag)
+            {
+                throw new Exception("Expected PING ACK");
```

Comment:
Why is it throwing `Exception`?  Usually these are `Http2ProtocolException` with a suitable error number etc.

Reply:
It's just test code.